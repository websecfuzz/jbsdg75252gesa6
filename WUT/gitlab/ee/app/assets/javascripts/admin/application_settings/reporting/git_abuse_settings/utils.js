import { parseBoolean } from '~/lib/utils/common_utils';

export const parseFormProps = ({
  maxNumberOfRepositoryDownloads,
  maxNumberOfRepositoryDownloadsWithinTimePeriod,
  gitRateLimitUsersAllowlist,
  gitRateLimitUsersAlertlist,
  autoBanUserOnExcessiveProjectsDownload,
}) => ({
  maxNumberOfRepositoryDownloads: parseInt(maxNumberOfRepositoryDownloads, 10),
  maxNumberOfRepositoryDownloadsWithinTimePeriod: parseInt(
    maxNumberOfRepositoryDownloadsWithinTimePeriod,
    10,
  ),
  gitRateLimitUsersAllowlist: JSON.parse(gitRateLimitUsersAllowlist),
  gitRateLimitUsersAlertlist: JSON.parse(gitRateLimitUsersAlertlist),
  autoBanUserOnExcessiveProjectsDownload: parseBoolean(autoBanUserOnExcessiveProjectsDownload),
});
