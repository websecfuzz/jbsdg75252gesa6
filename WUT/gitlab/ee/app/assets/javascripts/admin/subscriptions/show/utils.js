import {
  subscriptionTypes,
  offlineCloudLicenseText,
  onlineCloudLicenseText,
  licenseFileText,
} from './constants';

export const getLicenseFromData = ({ data } = {}) => data?.gitlabSubscriptionActivate?.license;
export const getErrorsAsData = ({ data } = {}) => data?.gitlabSubscriptionActivate?.errors || [];

export function getLicenseTypeLabel(type) {
  switch (type) {
    case subscriptionTypes.OFFLINE_CLOUD:
      return offlineCloudLicenseText;
    case subscriptionTypes.ONLINE_CLOUD:
      return onlineCloudLicenseText;
    default:
      return licenseFileText;
  }
}
