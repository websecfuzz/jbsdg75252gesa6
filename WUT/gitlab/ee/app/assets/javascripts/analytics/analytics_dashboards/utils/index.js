import produce from 'immer';
import { SESSIONS_TABLE_NAME } from 'ee/analytics/analytics_dashboards/constants';
import { DATE_RANGE_FILTER_DIMENSIONS } from 'ee/analytics/analytics_dashboards/data_sources/cube_analytics';
import getAllCustomizableDashboardsQuery from '../graphql/queries/get_all_customizable_dashboards.query.graphql';
import getCustomizableDashboardQuery from '../graphql/queries/get_customizable_dashboard.query.graphql';
import { TYPENAME_ANALYTICS_DASHBOARD_PANEL } from '../graphql/constants';

/**
 * Given a CubeJS property (e.g. `Sessions.count`), get the schema name (e.g. `Sessions`).
 */
export function getMetricSchema(metric) {
  return metric?.split('.')[0];
}

/**
 * Filters an array of dimensions by schema
 */
export function getDimensionsForSchema(selectedSchema, availableDimensions) {
  if (!selectedSchema) return [];

  return availableDimensions.filter(({ name }) => getMetricSchema(name) === selectedSchema);
}

/**
 * Selects a time dimension for a given schema
 */
export function getTimeDimensionForSchema(selectedSchema, availableTimeDimensions) {
  if (!selectedSchema) return null;

  const timeDimensions = availableTimeDimensions.filter(
    ({ name }) => getMetricSchema(name) === selectedSchema,
  );

  if (timeDimensions.length === 1) {
    // We only allow filtering by a single timeDimension. We expect most of our cubes to only have a single time dimension.
    return timeDimensions.at(0);
  }

  if (selectedSchema === SESSIONS_TABLE_NAME) {
    // Our `Sessions` cube is different, having both `startsAt`, `endsAt` timeDimensions.
    // We want to explicitly select the right dimension for Sessions so have this hardcoded lookup
    const sessionsDimensionName = DATE_RANGE_FILTER_DIMENSIONS[SESSIONS_TABLE_NAME.toLowerCase()];
    return timeDimensions.find(({ name }) => name === sessionsDimensionName);
  }

  // An unknown situation where a cube other than Sessions has more than one timeDimension. Hide it from the UI.
  return null;
}

/**
 * Updates a dashboard detail in cache from getProductAnalyticsDashboard:{slug}
 */
const updateDashboardDetailsApolloCache = ({
  apolloClient,
  dashboard,
  slug,
  fullPath,
  isProject,
  isGroup,
}) => {
  const getDashboardDetailsQuery = {
    query: getCustomizableDashboardQuery,
    variables: {
      fullPath,
      slug,
      isProject,
      isGroup,
    },
  };
  const sourceData = apolloClient.readQuery(getDashboardDetailsQuery);
  if (!sourceData) {
    // Dashboard details not yet in cache, must be a new dashboard, nothing to update
    return;
  }

  const data = produce(sourceData, (draftState) => {
    const { nodes } = isProject
      ? draftState.project.customizableDashboards
      : draftState.group.customizableDashboards;
    const updateIndex = nodes.findIndex((node) => node.slug === slug);

    if (updateIndex < 0) return;

    const updateNode = nodes[updateIndex];

    nodes.splice(updateIndex, 1, {
      ...updateNode,
      ...dashboard,
      panels: {
        ...updateNode.panels,
        nodes:
          dashboard.panels?.map((panel) => {
            const { id, ...panelRest } = panel;
            return { __typename: TYPENAME_ANALYTICS_DASHBOARD_PANEL, ...panelRest };
          }) || [],
      },
    });
  });

  apolloClient.writeQuery({
    ...getDashboardDetailsQuery,
    data,
  });
};

/**
 * Adds/updates a newly created dashboard to the dashboards list cache from getAllCustomizableDashboardsQuery
 */
const updateDashboardsListApolloCache = ({
  apolloClient,
  dashboardSlug,
  dashboard,
  fullPath,
  isProject,
  isGroup,
}) => {
  const getDashboardListQuery = {
    query: getAllCustomizableDashboardsQuery,
    variables: {
      fullPath,
      isProject,
      isGroup,
    },
  };
  const sourceData = apolloClient.readQuery(getDashboardListQuery);
  if (!sourceData) {
    // Dashboard list not yet loaded in cache, nothing to update
    return;
  }

  const data = produce(sourceData, (draftState) => {
    const { panels, ...dashboardWithoutPanels } = dashboard;
    const { nodes } = isProject
      ? draftState.project.customizableDashboards
      : draftState.group.customizableDashboards;

    const updateIndex = nodes.findIndex(({ slug }) => slug === dashboardSlug);

    // Add new dashboard if it doesn't exist
    if (updateIndex < 0) {
      nodes.push(dashboardWithoutPanels);
      return;
    }

    nodes.splice(updateIndex, 1, {
      ...nodes[updateIndex],
      ...dashboardWithoutPanels,
    });
  });

  apolloClient.writeQuery({
    ...getDashboardListQuery,
    data,
  });
};

export const updateApolloCache = ({
  apolloClient,
  slug,
  dashboard,
  fullPath,
  isProject,
  isGroup,
}) => {
  // TODO: modify to support removing dashboards from cache https://gitlab.com/gitlab-org/gitlab/-/issues/425513
  updateDashboardDetailsApolloCache({
    apolloClient,
    dashboard,
    slug,
    fullPath,
    isProject,
    isGroup,
  });
  updateDashboardsListApolloCache({ apolloClient, slug, dashboard, fullPath, isProject, isGroup });
};
