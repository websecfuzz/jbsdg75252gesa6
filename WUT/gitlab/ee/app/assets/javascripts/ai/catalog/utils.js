import { formValidators } from '@gitlab/ui/dist/utils';
import { s__, sprintf } from '~/locale';

export const createFieldValidators = ({ requiredLabel, maxLength } = {}) => {
  const validators = [];

  if (requiredLabel !== undefined) {
    validators.push(formValidators.required(requiredLabel));
  }

  if (maxLength !== undefined) {
    validators.push(
      formValidators.factory(
        sprintf(
          s__('AICatalog|Input cannot exceed %{value} characters. Please shorten your input.'),
          {
            value: maxLength,
          },
        ),
        (value) => (value?.length || 0) <= maxLength,
      ),
    );
  }

  return validators;
};
