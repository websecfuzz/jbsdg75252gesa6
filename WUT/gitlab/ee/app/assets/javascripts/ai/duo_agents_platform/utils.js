import { s__ } from '~/locale';
import { humanize } from '~/lib/utils/text_utility';

export const formatAgentDefinition = (agentDefinition) => {
  return humanize(agentDefinition || s__('DuoAgentsPlatform|Agent session'));
};

export const formatAgentFlowName = (agentDefinition, id) => {
  return `${formatAgentDefinition(agentDefinition)} #${id}`;
};

export const formatAgentStatus = (status) => {
  return status ? humanize(status.toLowerCase()) : s__('DuoAgentsPlatform|Unknown');
};
