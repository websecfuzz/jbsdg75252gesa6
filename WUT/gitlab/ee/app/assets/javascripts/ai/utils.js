import { duoChatGlobalState } from '~/super_sidebar/constants';
import { setCookie } from '~/lib/utils/common_utils';
import {
  DUO_AGENTIC_MODE_COOKIE,
  DUO_AGENTIC_MODE_COOKIE_EXPIRATION,
} from 'ee/ai/tanuki_bot/constants';

export const concatStreamedChunks = (arr) => {
  if (!arr) return '';

  let end = arr.findIndex((el) => !el);

  if (end < 0) end = arr.length;

  return arr.slice(0, end).join('');
};

/**
 * setCookie wrapper with duo agentic mode constances.
 *
 * @param {isAgenticMode} Boolean - Value to save
 * @returns {void}
 */
export const saveDuoAgenticModePreference = (isAgenticMode) => {
  setCookie(DUO_AGENTIC_MODE_COOKIE, isAgenticMode, {
    expires: DUO_AGENTIC_MODE_COOKIE_EXPIRATION,
  });
};

/**
 * Swith duo chat based on agenticMode value and save to cookie based on
 * saveCookie value.
 *
 * @param {agenticMode} Boolean - Agentic mode state
 * @param {saveCookie} Boolean - Save to cookie flag
 * @returns {void}
 */

export const setAgenticMode = (agenticMode = true, saveCookie = false) => {
  duoChatGlobalState.isShown = !agenticMode;
  duoChatGlobalState.isAgenticChatShown = agenticMode;

  if (saveCookie) {
    saveDuoAgenticModePreference(agenticMode);
  }
};

/**
 * Sends a command to DuoChat to execute on. This should be use for
 * a single command.
 *
 * @param {question} String - Prompt to send to the chat endpoint
 * @param {resourceId} String - Unique ID to bind the streaming
 * @param {variables} Object - Additional variables to pass to graphql chat mutation
 */
export const sendDuoChatCommand = ({ question, resourceId, variables = {} } = {}) => {
  if (!question || !resourceId) {
    throw new Error('Both arguments `question` and `resourceId` are required');
  }
  setAgenticMode(false, true);

  window.requestIdleCallback(() => {
    duoChatGlobalState.commands.push({
      question,
      resourceId,
      variables,
    });
  });
};

export const clearDuoChatCommands = () => {
  duoChatGlobalState.commands = [];
};

/**
 * Converts a text string into a URL-friendly format for event tracking.
 *
 * - Converts to lowercase
 * - Removes special characters
 * - Replaces spaces with underscores
 * - Limits length to 50 characters
 *
 * @param {string} text - The text to convert
 * @returns {string} The formatted event label
 */
export const generateEventLabelFromText = (text) => {
  return text
    .toLowerCase()
    .replace(/[^a-z0-9\s]/g, '')
    .replace(/\s+/g, '_')
    .substring(0, 50);
};

export const utils = {
  concatStreamedChunks,
  generateEventLabelFromText,
};
