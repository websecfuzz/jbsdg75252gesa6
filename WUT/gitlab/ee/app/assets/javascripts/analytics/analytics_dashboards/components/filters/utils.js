import { queryToObject } from '~/lib/utils/url_utility';
import { formatDate, newDate } from '~/lib/utils/datetime_utility';
import { ISO_SHORT_FORMAT } from '~/vue_shared/constants';
import {
  convertObjectPropsToCamelCase,
  convertObjectPropsToSnakeCase,
  parseBoolean,
} from '~/lib/utils/common_utils';
import {
  filterToQueryObject,
  urlQueryToFilter,
} from '~/vue_shared/components/filtered_search_bar/filtered_search_utils';
import {
  START_DATES,
  DATE_RANGE_OPTIONS,
  DATE_RANGE_OPTION_CUSTOM,
  DATE_RANGE_OPTION_KEYS,
  DEFAULT_SELECTED_DATE_RANGE_OPTION,
  FILTERED_SEARCH_SUPPORTED_TOKENS,
  PROJECT_FILTER_QUERY_NAME,
} from './constants';

const isCustomOption = (option) => option && option === DATE_RANGE_OPTION_CUSTOM;

export const getDateRangeOption = (optionKey) => DATE_RANGE_OPTIONS[optionKey] || null;

export const dateRangeOptionToFilter = ({ startDate, endDate, key }) => ({
  startDate,
  endDate,
  dateRangeOption: key,
});

const DEFAULT_FILTER = dateRangeOptionToFilter(
  DATE_RANGE_OPTIONS[DEFAULT_SELECTED_DATE_RANGE_OPTION],
);

export const buildDefaultDashboardFilters = (queryString, dashboardDefaultFilters = {}) => {
  const {
    dateRangeOption,
    startDate,
    endDate,
    filterAnonUsers,
    [PROJECT_FILTER_QUERY_NAME]: projectFullPath,
  } = convertObjectPropsToCamelCase(queryToObject(queryString, { gatherArrays: true }));
  const searchFilters = urlQueryToFilter(queryString, {
    filterNamesAllowList: FILTERED_SEARCH_SUPPORTED_TOKENS,
  });

  const isDateRangeFilterEnabled = dashboardDefaultFilters?.dateRange?.enabled;
  const optionKey = dateRangeOption || dashboardDefaultFilters?.dateRange?.defaultOption;
  const dateRangeOverride = DATE_RANGE_OPTION_KEYS.includes(optionKey)
    ? dateRangeOptionToFilter(getDateRangeOption(optionKey))
    : {};

  const customDateRange = isCustomOption(optionKey);

  return {
    ...(isDateRangeFilterEnabled && {
      ...DEFAULT_FILTER,
      // Override default filter with user defined option
      ...dateRangeOverride,
      // Override date range when selected option is custom date range
      ...(customDateRange && { startDate: newDate(startDate) }),
      ...(customDateRange && { endDate: newDate(endDate) }),
    }),
    filterAnonUsers: parseBoolean(filterAnonUsers),
    searchFilters,
    projectFullPath: projectFullPath || null,
  };
};

export const filtersToQueryParams = ({
  dateRangeOption,
  startDate,
  endDate,
  filterAnonUsers,
  searchFilters,
  projectFullPath,
}) => {
  const customDateRange = isCustomOption(dateRangeOption);

  const searchFiltersQueryObj = filterToQueryObject(searchFilters);
  const additionalFiltersQueryObj = convertObjectPropsToSnakeCase({
    dateRangeOption,
    // Clear the date range unless the custom date range is selected
    startDate: customDateRange ? formatDate(startDate, ISO_SHORT_FORMAT) : null,
    endDate: customDateRange ? formatDate(endDate, ISO_SHORT_FORMAT) : null,
    // Clear the anon users filter unless truthy
    filterAnonUsers: filterAnonUsers || null,
    [PROJECT_FILTER_QUERY_NAME]: projectFullPath || null,
  });

  return { ...additionalFiltersQueryObj, ...searchFiltersQueryObj };
};

export function isDashboardFilterEnabled(filter) {
  return filter?.enabled || false;
}

export const getStartDate = (option) =>
  START_DATES[option] ?? START_DATES[DEFAULT_SELECTED_DATE_RANGE_OPTION];
