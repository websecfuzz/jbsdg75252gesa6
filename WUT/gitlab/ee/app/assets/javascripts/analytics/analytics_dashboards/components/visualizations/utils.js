import { isNil } from 'lodash';
import { isNumeric } from '~/lib/utils/number_utils';
import { formatNumber, n__, __, sprintf } from '~/locale';
import { formatDate, humanizeTimeInterval } from '~/lib/utils/datetime/date_format_utility';
import { formatAsPercentage } from 'ee/analytics/dora/components/util';
import { NULL_SERIES_ID } from 'ee/analytics/shared/constants';
import { UNITS } from '~/analytics/shared/constants';

function isIsoDateString(dateString) {
  // Matches an ISO date string in the format `2024-03-14T00:00:00.000`
  const isoDateRegex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}$/;
  return isoDateRegex.test(dateString);
}

export function formatVisualizationValue(value) {
  if (isIsoDateString(value)) {
    return formatDate(value);
  }

  if (isNumeric(value)) {
    return formatNumber(parseInt(value, 10));
  }

  return value;
}

export function formatVisualizationTooltipTitle(title, params) {
  const value = params?.seriesData?.at(0)?.value?.at(0);

  if (isIsoDateString(value)) {
    const formattedDate = formatDate(value);
    return title.replace(value, formattedDate);
  }

  return title;
}

export function customFormatVisualizationTooltipTitle(params, formatter) {
  const xAxisValue = params?.seriesData?.at(0)?.value?.at(0);

  if (isNil(xAxisValue)) return '';

  return formatter(xAxisValue);
}

export const humanizeDisplayUnit = ({ unit, data = 0 }) => {
  switch (unit) {
    case 'days':
      return n__('day', 'days', data === '-' ? 0 : data);
    case 'per_day':
      return __('/day');
    case 'percent':
      return '%';
    default:
      return unit;
  }
};

/**
 * Humanizes values to be displayed in chart tooltips
 *
 * @param {string} unit â€“ The unit of measurement to be used for metric
 * @param {number} value - The value of the metric
 * @returns {string|number} - Humanized tooltip value
 */
export const humanizeChartTooltipValue = ({ unit, value } = {}) => {
  if (isNil(value)) return __('No data');

  switch (unit) {
    case UNITS.DAYS:
      return n__('%d day', '%d days', value);
    case UNITS.PER_DAY:
      return sprintf(__('%{value} /day'), { value });
    case UNITS.PERCENT:
      return formatAsPercentage(value);
    case UNITS.TIME_INTERVAL:
      return humanizeTimeInterval(value);
    default:
      return value;
  }
};

export const calculateDecimalPlaces = ({ data, decimalPlaces } = {}) => {
  return (data && parseInt(decimalPlaces, 10)) || 0;
};

export const removeNullSeries = (seriesData) => {
  return seriesData?.filter(({ seriesId }) => seriesId !== NULL_SERIES_ID);
};
