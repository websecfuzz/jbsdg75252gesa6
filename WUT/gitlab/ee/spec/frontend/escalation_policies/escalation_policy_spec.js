import { GlSprintf, GlIcon, GlCard, GlToken } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import { cloneDeep } from 'lodash';
import EditEscalationPolicyModal from 'ee/escalation_policies/components/add_edit_escalation_policy_modal.vue';
import DeleteEscalationPolicyModal from 'ee/escalation_policies/components/delete_escalation_policy_modal.vue';
import EscalationPolicy, { i18n } from 'ee/escalation_policies/components/escalation_policy.vue';

import {
  deleteEscalationPolicyModalId,
  editEscalationPolicyModalId,
} from 'ee/escalation_policies/constants';
import { parsePolicy } from 'ee/escalation_policies/utils';
import mockPolicies from './mocks/mockPolicies.json';

describe('EscalationPolicy', () => {
  let wrapper;
  const escalationPolicy = parsePolicy(cloneDeep(mockPolicies[0]));
  const escalationPolicyWithoutUsers = parsePolicy(cloneDeep(mockPolicies[1]));

  const createComponent = (policy = escalationPolicy) => {
    wrapper = shallowMount(EscalationPolicy, {
      propsData: {
        policy,
        index: 0,
      },
      stubs: {
        GlSprintf,
        GlCard,
      },
    });
  };

  beforeEach(() => {
    createComponent();
  });

  const findDeleteModal = () => wrapper.findComponent(DeleteEscalationPolicyModal);
  const findEditModal = () => wrapper.findComponent(EditEscalationPolicyModal);
  const findWarningIcon = () => wrapper.findComponent(GlIcon);
  const findGlTokens = () => wrapper.findAllComponents(GlToken);

  it('renders a policy with rules', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a policy without rules', () => {
    const policyWithoutRules = {
      id: mockPolicies[0].id,
      name: mockPolicies[0].name,
      description: mockPolicies[0].description,
      rules: [],
    };
    createComponent(policyWithoutRules);
    expect(findWarningIcon().exists()).toBe(true);
    expect(wrapper.text()).toContain(i18n.noRules);
  });

  describe('Modals', () => {
    describe('delete policy modal', () => {
      it('should render a modal and provide it with correct id', () => {
        const modal = findDeleteModal();
        expect(modal.exists()).toBe(true);
        expect(modal.props('modalId')).toBe(
          `${deleteEscalationPolicyModalId}-${escalationPolicy.id}`,
        );
      });
    });

    describe('edit policy modal', () => {
      it('should render a modal and provide it with correct id and isEditMode props', () => {
        const modal = findEditModal();
        expect(modal.exists()).toBe(true);
        expect(modal.props('modalId')).toBe(
          `${editEscalationPolicyModalId}-${escalationPolicy.id}`,
        );
        expect(modal.props('isEditMode')).toBe(true);
      });
    });
  });

  describe('User tokens', () => {
    it('do not render when escalation rule has no assigned users', () => {
      createComponent(escalationPolicyWithoutUsers);
      expect(findGlTokens()).toHaveLength(0);
    });

    it('render for all mapped participants', () => {
      expect(findGlTokens()).toHaveLength(wrapper.vm.mappedParticipants.length);
    });

    it('have assigned style and class attributes from mapped participants', () => {
      findGlTokens().wrappers.forEach((token) => {
        expect(token.attributes('style')).toContain('background-color');
        expect(token.attributes('class')).toContain('gl-text-white');
      });
    });

    it('have distinctive color for each participant in a rule', () => {
      const tokensStyleAttribute = findGlTokens().wrappers.map((w) => w.attributes('style'));
      expect(tokensStyleAttribute[0]).not.toBe(tokensStyleAttribute[1]);
    });
  });
});
