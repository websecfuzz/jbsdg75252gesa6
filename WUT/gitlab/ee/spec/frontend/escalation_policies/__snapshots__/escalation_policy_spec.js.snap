// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EscalationPolicy renders a policy with rules 1`] = `
<div>
  <div
    class="gl-card gl-mt-5"
  >
    <div
      class="gl-card-header gl-py-3"
    >
      <div
        class="gl-flex gl-items-center"
      >
        <gl-button-stub
          aria-label="Collapse"
          buttontextclasses=""
          category="tertiary"
          class="!gl-p-0 gl-mr-2"
          icon=""
          size="medium"
          tag="button"
          title="Collapse"
          type="button"
          variant="default"
        >
          <gl-icon-stub
            name="chevron-lg-down"
            size="12"
            variant="current"
          />
        </gl-button-stub>
        <h3
          class="gl-font-bold gl-m-0 gl-text-lg"
        >
          Escalation policy
        </h3>
        <gl-button-group-stub
          class="gl-ml-auto"
        >
          <gl-button-stub
            aria-label="Edit escalation policy"
            buttontextclasses=""
            category="primary"
            icon="pencil"
            role="button"
            size="medium"
            tabindex="0"
            tag="button"
            title="Edit escalation policy"
            type="button"
            variant="default"
          />
          <gl-button-stub
            aria-label="Delete escalation policy"
            buttontextclasses=""
            category="primary"
            icon="remove"
            role="button"
            size="medium"
            tabindex="0"
            tag="button"
            title="Delete escalation policy"
            type="button"
            variant="default"
          />
        </gl-button-group-stub>
      </div>
    </div>
    <div
      class="gl-card-body gl-p-0"
    >
      <gl-collapse-stub
        tag="div"
        visible="true"
      >
        <div
          class="gl-p-5"
        >
          <p
            class="gl-mb-5 gl-text-subtle"
          >
            Description 1 lives here
          </p>
          <div
            class="gl-border-1 gl-border-default gl-border-solid gl-p-5 gl-rounded-base"
          >
            <div
              class="escalation-rule-row gl-flex gl-items-center gl-mb-5"
            >
              <span
                class="md:gl-w-full rule-condition"
              >
                <gl-icon-stub
                  class="gl-mr-3"
                  name="clock"
                  size="16"
                  variant="current"
                />
                IF alert is not acknowledged in
                <span
                  class="gl-font-bold"
                >
                  1 mins
                </span>
              </span>
              <span
                class="gl-bg-gray-900 gl-hidden gl-min-w-7 gl-mx-5 gl-relative gl-shrink-0 lg:gl-block right-arrow"
                style="width: 4%;"
              >
                <i
                  class="gl-absolute gl-border-solid gl-p-1 right-arrow-head"
                />
              </span>
              <span
                class="gl-flex gl-items-center gl-min-w-0"
              >
                <span
                  class="gl-truncate"
                  title=" THEN email on-call user in schedule Schedule to fill in"
                >
                  <gl-icon-stub
                    class="gl-mr-3"
                    name="notifications"
                    size="16"
                    variant="current"
                  />
                  THEN email on-call user in schedule
                  <span
                    class="gl-font-bold"
                  >
                    Schedule to fill in
                  </span>
                </span>
              </span>
            </div>
            <div
              class="escalation-rule-row gl-flex gl-items-center gl-mb-5"
            >
              <span
                class="md:gl-w-full rule-condition"
              >
                <gl-icon-stub
                  class="gl-mr-3"
                  name="clock"
                  size="16"
                  variant="current"
                />
                IF alert is not resolved in
                <span
                  class="gl-font-bold"
                >
                  2 mins
                </span>
              </span>
              <span
                class="gl-bg-gray-900 gl-hidden gl-min-w-7 gl-mx-5 gl-relative gl-shrink-0 lg:gl-block right-arrow"
                style="width: 8%;"
              >
                <i
                  class="gl-absolute gl-border-solid gl-p-1 right-arrow-head"
                />
              </span>
              <span
                class="gl-flex gl-items-center gl-min-w-0"
              >
                <span
                  class="gl-truncate"
                  title=" THEN email user Lena"
                >
                  <gl-icon-stub
                    class="gl-mr-3"
                    name="notifications"
                    size="16"
                    variant="current"
                  />
                  THEN email user
                  <gl-token-stub
                    class="gl-text-white"
                    removelabel="Remove"
                    style="background-color: rgb(97, 122, 226);"
                    variant="default"
                    viewonly="true"
                  >
                    <gl-avatar-stub
                      alt="avatar"
                      entityid="0"
                      entityname=""
                      shape="circle"
                      size="16"
                      src="avatar.com/lena.png"
                    />
                    Lena
                  </gl-token-stub>
                </span>
              </span>
            </div>
            <div
              class="escalation-rule-row gl-flex gl-items-center"
            >
              <span
                class="md:gl-w-full rule-condition"
              >
                <gl-icon-stub
                  class="gl-mr-3"
                  name="clock"
                  size="16"
                  variant="current"
                />
                IF alert is not resolved in
                <span
                  class="gl-font-bold"
                >
                  4 mins
                </span>
              </span>
              <span
                class="gl-bg-gray-900 gl-hidden gl-min-w-7 gl-mx-5 gl-relative gl-shrink-0 lg:gl-block right-arrow"
                style="width: 12%;"
              >
                <i
                  class="gl-absolute gl-border-solid gl-p-1 right-arrow-head"
                />
              </span>
              <span
                class="gl-flex gl-items-center gl-min-w-0"
              >
                <span
                  class="gl-truncate"
                  title=" THEN email user Tim"
                >
                  <gl-icon-stub
                    class="gl-mr-3"
                    name="notifications"
                    size="16"
                    variant="current"
                  />
                  THEN email user
                  <gl-token-stub
                    class="gl-text-white"
                    removelabel="Remove"
                    style="background-color: rgb(201, 93, 46);"
                    variant="default"
                    viewonly="true"
                  >
                    <gl-avatar-stub
                      alt="avatar"
                      entityid="0"
                      entityname=""
                      shape="circle"
                      size="16"
                      src="avatar.com/tim.png"
                    />
                    Tim
                  </gl-token-stub>
                </span>
              </span>
            </div>
          </div>
        </div>
      </gl-collapse-stub>
    </div>
  </div>
  <delete-escalation-policy-modal-stub
    escalationpolicy="[object Object]"
    modalid="deleteEscalationPolicyModal-37"
  />
  <edit-escalation-policy-modal-stub
    escalationpolicy="[object Object]"
    iseditmode="true"
    modalid="editEscalationPolicyModal-37"
  />
</div>
`;
